### ISSUE TITLE
[audit] Set purged_size=21 in walk-forward CV presets — remove train/test contamination

### ISSUE THREAD

**Issue by SilvioBaratto** [SilvioBaratto/optimizer#270]
## Problem

`WalkForwardConfig` defaults to `purged_size=0` — no temporal buffer between training and test windows. This allows information leakage from the end of the training set into the beginning of the test set, inflating cross-validation scores. Best practice is at least 1 trading week (5 days) of purging; 21 days (1 month) is more conservative for monthly-rebalanced strategies.

## Current State

`optimizer/validation/_config.py:37`:

```python
purged_size: int = 0
```

`for_quarterly_rolling()` (line 49) does not override this:

```python
def for_quarterly_rolling(cls) -> WalkForwardConfig:
    return cls(test_size=63, train_size=252)  # no purged_size set
```

## Acceptance Criteria

- [ ] Set `purged_size=21` in `WalkForwardConfig.for_quarterly_rolling()`
- [ ] Consider setting a sensible default in the base config (e.g., `purged_size=5`)
- [ ] Add a note in the docstring explaining the purging rationale
- [ ] Verify existing tests still pass with the new default

## Files to Modify

- `optimizer/validation/_config.py` — update `for_quarterly_rolling()` and optionally the default

## Dependencies

None

## Estimated Impact

Sharpe drops ~0.05-0.10 (removes temporal leakage bias)

## Priority

P2 — Significant: standard practice for temporal CV, omission inflates reported performance
